CMakeLists.txt

```diff
@@ -1,83 +1,191 @@
+find_package(pip)
+
+# Check if OpenMPI is present
+find_package(MPI REQUIRED)
+if (MPI_NOT_FOUND)
+    MESSAGE(FATAL_ERROR "We did not found any OpenMPI installation. Please run ./prerequisites.sh again before
+calling cmake. The OpenMPI library is required in order to successfully use BioDynaMo." )
 endif()
 
-add_definitions(-DBDM_SRC_DIR=\"${CMAKE_SOURCE_DIR}/src\")  # escaped quotes are mandatory
+# Check if GLUT is present. If we do not have it, then we
+# disable directly paraview.
+find_package(GLUT)
+if(NOT GLUT_FOUND)
+    PRINT_WARNING()
+    MESSAGE("GLUT libraries were not found in the system. In order to use ParaView they need to be installed.
+If you want to do that automatically, please run the ./prerequisites.sh script.")
+    PRINT_LINE()
+    SET(paraview OFF)
+endif()
 
-# -------------------- find packages ------------------------------------------
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
-# CMake cannot find LLVM's Clang OpenMP library by default (this makes
-# find_package(OpenMP) fail). Therefore we manually specify the OpenMP flags.
-# Furthermore, rootcling cannot find omp.h by default, so we copy this (with a
-# shell script to /usr/local/Cellar/biodynamo, which is brew's default install
-# directory), and include this directory. We cannot directly include the
-# original directory, because of header conflicts (such as stdint.h)
-if(APPLE)
-  link_directories(/usr/local/opt/llvm/lib)
-  set(OpenMP_C_FLAGS -fopenmp=libomp)
-  set(OpenMP_CXX_FLAGS -fopenmp=libomp)
-  include_directories(/usr/local/Cellar/biodynamo)
+# Check if we have a compatible openmp compiler
+find_package(OpenMP)
+if (OpenMP_FOUND)
+    link_directories(${OpenMP_CXX_LIBRARIES})
 else()
-  find_package(OpenMP REQUIRED)
+    MESSAGE(FATAL_ERROR "OpenMP was not found in your system. Please run ./prerequisites.sh again before
+calling cmake. If this does not work, then your current compiler might not support OpenMP (this happens for instance
+with Apple's standard compiler). Please update it to a more recent version and then run cmake again.")
 endif()
+
 if (test)
   # must be before include(${ROOT_USE_FILE}) otherwise test compile fails
-  find_package(Threads REQUIRED)
+  find_package(Threads)
 endif()
 
 if(cuda)
   find_package(CUDA)
+  if (NOT CUDA_FOUND)
+    MESSAGE("CUDA was not found in your system, therefore it was disabled to let you compile BioDynaMo. In order to
+enable this functionality please install CUDA for your system. Please have a look at https://docs.nvidia.com/cuda/index.html.
+Then run again cmake (with the -Dcuda=ON flag).")
+    SET(cuda OFF)
+  endif()
 endif()
+
 if(opencl)
   find_package(OpenCL)
+  if (NOT OpenCL_FOUND)
+      MESSAGE("OpenCL was not found in your system, therefore it was disabled to let you compile BioDynaMo. In order to
+enable this functionality please install OpenCL for your system. Please have a look at https://developer.nvidia.com/opencl.
```

Comment:
Please have a look at the website of your graphic card vendor on how to install OpenCL.